students/JonathanMauk/lesson05/mailroom03.py

```diff
@@ -0,0 +1,148 @@
+            total_given = sum(donor_db[k])
+            average_gifts = total_given / num_gifts
+            print(f'{k: <26}| ${total_given:>10.2f} |{num_gifts:^11}| ${average_gifts:>11.2f}')
+        print('\nReturning to main menu...')
+        return
+
+
+def quit_program():
+    """Quit Mailroom program."""
+    print("Exiting...")
+    sys.exit()
+
+
+def create_letter(donor_status, donor_name, donation_amt):
+    """Return and print formatted letters, depending on options selected. Not intended to be used by itself."""
+    if donor_status == 0:
+        letter_text = """
+        Dear {0},
+    
+            Thank you for your very kind donation of ${1}, and for your continuing support.
+    
+            Your generous contribution will be put to very good use.
+    
+                           Sincerely,
+                              -The Team
+                              """.format(donor_name, donation_amt)
+        print(letter_text)
+        print("---------------------------")
+        print("Returning to thank you letter menu...")
+        thank_you()
+    elif donor_status == 1:
+        letter_text = """
+        Dear {0},
+
+            Thank you for your very kind donation of ${1}.
+
+            Your generous contribution will be put to very good use.
+
+                           Sincerely,
+                              -The Team
+                              """.format(donor_name, donation_amt)
+        print(letter_text)
+        print("---------------------------")
+        print("Returning to thank you letter menu...")
+        thank_you()
+    elif donor_status == 2:
+        return("""
+        Dear {0},
+
+            Thank you for your very kind contribution(s) totaling ${1}.
+
+            We would like you to know that your generous donation(s) will be put to very good use.
+
+                           Sincerely,
+                              -The Team
+                              """.format(donor_name, donation_amt))
+
+
+def thank_all():
+    """Write letters generated by create_letter to text files, saving them to same directory as script."""
+    current_dir = os.getcwd()
+    print("Saving letters to {0}...".format(current_dir))
+
+    for k, v in donor_db.items():
+        letter = create_letter(2, k, sum(v))
+        with open('{:s}.txt'.format(k), 'w') as f:
+            f.write(letter)
+    print("---------------------------")
+    print("Letters saved to text files in directory. Returning to main menu...")
+    mailroom()
+
+
+def mailroom():
+    """Generate main menu options and activate other functions."""
+    while True:
+        selection = input('MAILROOM v0.3\n------------------------\nChoose an option:\n1) Send a thank you' +
+                          '\n2) Create a report\n3) Send letters to everyone\n4) Quit\n> ')
+        menu_dict = {'1': thank_you, '2': report, '3': thank_all, '4': quit_program}
+        try:
+            menu_dict.get(selection)()
```

Comment:
An alternative would be to use the bracket syntax to access dict values and then catch a `KeyError` for bad input